temporal-kotlin/src/main/kotlin/io/temporal/common/converter/KotlinObjectMapperFactory.kt

```diff
@@ -29,9 +29,14 @@ class KotlinObjectMapperFactory {
     fun new(): ObjectMapper {
       val mapper = JacksonJsonPayloadConverter.newDefaultObjectMapper()
 
-      // use deprecated constructor instead of builder to maintain compatibility with old jackson versions
-      @Suppress("deprecation")
-      val km = KotlinModule()
+      val km = try {
+        KotlinModule.Builder()
+          .build()
```

Comment:
Have we confirmed this works with jackson `v2.9.0`?

Reply:
The existing code doesn't work with jackson `2.9.0`

Reply:
Not sure I understand, the Java SDK currently does build and all test pass with jackson `2.9.0`

Reply:
It currently builds with jackson 1.14.2 https://github.com/temporalio/sdk-java/blob/5ccb85927acf01db619eea4eefae0fb41c5ad471/build.gradle#L34

Reply:
It fails with jackson 2.9.0
```
'void com.fasterxml.jackson.module.kotlin.KotlinModule.<init>(int, boolean, boolean, boolean, com.fasterxml.jackson.module.kotlin.SingletonSupport, boolean, int, kotlin.jvm.internal.DefaultConstructorMarker)'
java.lang.NoSuchMethodError: 'void com.fasterxml.jackson.module.kotlin.KotlinModule.<init>(int, boolean, boolean, boolean, com.fasterxml.jackson.module.kotlin.SingletonSupport, boolean, int, kotlin.jvm.internal.DefaultConstructorMarker)'
```

Reply:
Note the comment indicating our support for `2.9.0`

Reply:
Unless I'm doing something wrong, the code as-is won't work with jackson 2.9.0 https://github.com/temporalio/sdk-java/compare/master...Sineaggi:sdk-java:jackson-version-test
If you run `./gradlew :temporal-kotlin:test`

Reply:
If we need to drop support to maintain forward compatibility I would 100% be in favor, but if we can support older versions without harming forward compatibility that would be the best for all SDK users.

Reply:
I've gone the `registerKotlinModule` route, which should allow for maximum backwards and forwards compatibility.